internal/log/access.go

```diff
@@ -19,18 +24,48 @@ const (
 	AccessLogFieldUserAgent           AccessLogField = "user-agent"
 )
 
-// DefaultAccessLogFields are the default access log fields to log.
-var DefaultAccessLogFields = []AccessLogField{
-	AccessLogFieldUpstreamCluster,
-	AccessLogFieldMethod,
-	AccessLogFieldAuthority,
-	AccessLogFieldPath,
-	AccessLogFieldUserAgent,
-	AccessLogFieldReferer,
-	AccessLogFieldForwardedFor,
-	AccessLogFieldRequestID,
-	AccessLogFieldDuration,
-	AccessLogFieldSize,
-	AccessLogFieldResponseCode,
-	AccessLogFieldResponseCodeDetails,
+// DefaultAccessLogFields returns the default access log fields.
+func DefaultAccessLogFields() []AccessLogField {
+	return []AccessLogField{
+		AccessLogFieldUpstreamCluster,
+		AccessLogFieldMethod,
+		AccessLogFieldAuthority,
+		AccessLogFieldPath,
+		AccessLogFieldUserAgent,
+		AccessLogFieldReferer,
+		AccessLogFieldForwardedFor,
+		AccessLogFieldRequestID,
+		AccessLogFieldDuration,
+		AccessLogFieldSize,
+		AccessLogFieldResponseCode,
+		AccessLogFieldResponseCodeDetails,
+	}
+}
+
+// ErrUnknownAccessLogField indicates that an access log field is unknown.
+var ErrUnknownAccessLogField = errors.New("unknown access log field")
+
+var accessLogFieldLookup = map[AccessLogField]struct{}{
```

Comment:
It would be nice if there were some way to generate this map automatically, so we didn't have to worry about keeping it in sync with the definitions above, but I can't think of anything easy at the moment.

(no action required)

Reply:
We could use the list we made for defaults, but when we add new fields they won't be part of the defaults, so we wouldn't be able to use that slice anymore.